packages/block-library/src/navigation/edit/use-create-navigation-menu.js

```diff
@@ -57,14 +58,47 @@ export default function useCreateNavigationMenu( clientId ) {
 					);
 				} );
 			}
+
 			const record = {
 				title,
+				slug,
```

Comment:
Passing slug here allows us to save navigation blocks using their position within the template part hierarchy encoded into the slug (e.g. `header--header-small`). 

In the future we can use this to fetch the most appropriate Navigation Menu based on the slug that most closely correlates with the slugified version of block's position within the template hierarchy